modules/tensor_mechanics/include/materials/DiscreteRRIsotropicPlasticity.h

```diff
@@ -0,0 +1,62 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef DISCRETERRISOTROPICPLASTICITY_H
+#define DISCRETERRISOTROPICPLASTICITY_H
+
+#include "DiscreteRadialReturnStressIncrement.h"
+#include "PiecewiseLinear.h"
+#include "MooseMesh.h"
+
+// forward declarations
+class PiecewiseLinear;
+
+/**
+ */
+
+class DiscreteRRIsotropicPlasticity : public DiscreteRadialReturnStressIncrement
+{
+public:
+  DiscreteRRIsotropicPlasticity( const InputParameters & parameters);
+
+protected:
+  void resetQpProperties(); // Values here need to be set to a constant, ideally zero
+
+  virtual void computeStressInitialize(Real effectiveTrialStress);
+  virtual Real computeResidual(Real effectiveTrialStress, Real scalar);
+  virtual Real computeDerivative(Real effectiveTrialStress, Real scalar);
+  virtual void iterationFinalize(Real scalar);
+  virtual void computeStressFinalize(const RankTwoTensor & plasticStrainIncrement);
```

Comment:
We do need to standardize. I propose that we start making use of some of the more advanced features of Doxygen:

There is an [http://www.stack.nl/~dimitri/doxygen/manual/config.html#cfg_inherit_docs](INHERIT_DOCS tag) that can be used to tell Doxygen to inherit documentation of virtual members. This really seems like a good idea. There's no reason for developers to waste time redefining things like `computeResidual` and `computeJacobian` all the way down their inheritance chains. 

I think we do need to comment all new methods introduced in classes. As far as class variables go, that may be a good idea too... Let's see if we can get the method documentation standardized first. One step at a time.